node/network/availability-distribution/src/lib.rs

```diff
@@ -503,12 +456,11 @@ where
 	}
```

Comment:
This `peers.is_empty()` has the same issue a the newly introduced one.

The following must be called at the very least for each item provided by `messages`.
```rust
		per_candidate
			.message_vault
			.insert(message.erasure_chunk.index, message.clone());
```

Reply:
Thanks. This is what I was referencing with my other comment.